### DISCUSSION TITLE
Can't set working directory for building site

### DISCUSSION THREAD

**Discussion by xenilen** [retypeapp/retype#290] [Q&A]
Hello!

I'm trying to configure retypeapp/action-build and retypeapp/action-github-pages to only build files from a specific directory/sub-directories.  However, it seems like there isn't a feature present to change the working directory.  Maybe there is and I am doing something wrong.

Example site setup:
/ (Where your main site is)
/wiki (Where I want the retypeapp to live)

The only thing I've found is 

    defaults:
        run:
            working-directory: wiki

But this doesn't apply to `uses:`

**Comment by geoffreymcgill** [comment]
Is your repo public? Can you share a link to the repo?

If your projects `retype.yml` file is within a sub-folder and not within the root of the repo, you can configure the path:

https://github.com/retypeapp/action-build#specify-path-to-the-retypeyml-file

```yml
- uses: retypeapp/action-build@v1
  with:
    config: wiki
```

Hope this helps.

**Comment by xenilen** [comment]
Thanks for the quick reply

> If your projects retype.yml file is within a sub-folder and not within the root of the repo, you can configure the path:

I've played around with this already.  If I understand it correctly, it takes the retype.yml file in the targeted directory, generates the retype html files, and creates the retype branch with only that information.  Which isn't really what I was looking for.  

Maybe I'm just miss-understanding how to utilize the stuff in the uses: field.  I'm now thinking that I should make the action generate the retype portion of the site, then merge it with the normal html portion?

My repo is here [https://github.com/xenilen/xenilen.github.io](https://github.com/xenilen/xenilen.github.io)

My goal is to have /index.html be my homepage, separate from the retype wiki located in /redteam_cookbook/index.html

Also, it's kind of confusing that the name retype.yml is used multiple times for different things in the docs ie [here](https://retype.com/guides/github-actions/#step-1-add-retypeyml-workflow) and [here](https://retype.com/configuration/project/)

**Reply by geoffreymcgill** [reply]
> I'm now thinking that I should make the action generate the retype portion of the site, then merge it with the normal html portion?

Yes, exactly. 

The primary challenge here is the `.github/workflows/retype.yml` builds the Retype portion of your website and by default copies the generated .html files into a new `retype` branch. That portion of the build process is working correctly and you can see the generated files within https://github.com/xenilen/xenilen.github.io/tree/retype.

But, you require one of the following options to happen:

1. Copy the Retype generated files into the `main` branch, or
2. Copy your custom .html files into the `retype` branch

Personally, I feel option 2 above would be better. That keeps your content source files separated (mostly) from the .html files used by the webserver. 

I'll have to do a few experiments to try and get this working. Another team member that I would leverage for assistance is offline until Monday, but I'll see if I can make it work. 

> Also, it's kind of confusing that the name retype.yml is used multiple times for different things in the docs ie [here](https://retype.com/guides/github-actions/#step-1-add-retypeyml-workflow) and [here](https://retype.com/configuration/project/)

Yes, agreed. I'm going to fix that today. We will rename the `.github/workflows/retype.yml` file to something else. Maybe `.github/workflows/retype-action.yml`.

**Comment by fabriciomurta** [comment]
I can see three options here, in order of complexity:

### Simplest, all in same branch, hide retype sources by folder.

1. Change github pages to the `main` branch to, instead of `/` to a subdir, say `/website`
2. Create the directory and move all your files from the current website therein, including CNAME.
3. Then create another dir, say, `/docs`. Then `retype.yml` project configuration file and all your document pages (.md) for documentation assembled by Retype
4. GitHub workflow (`.github/workflows/retype.yml`) **action-build** will have `config: docs/` argument
5. Workflow **action-github-pages** will have `branch: main` and `directory: website/wiki`

And that should do it. You may want to set up the workflow to trigger only if the changes pushed affect the `/docs/**` path in the repository; changes in files outside `docs` should not result in different Retype output.

### A little more elaborate, same GitHubPages setting, retype files in another (orphan) branch

1. Checkout an orphan GitHub repository say, `retype-source`
2. Move all Retype files, including the workflow file (once you create an orphan branch, you'll have all files from last branch -- `main` -- staged for commit; remove all that are not related to Retype, commit)
3. Remove Retype and workflow files from the `main` branch
4. The `action-build` step now needs no extra arguments
5. The `action-github-pages` will need `branch: main` and `directory: wiki` arguments

Then every time you push changes to Retype, or `retype-source` branch, the workflow will run and update files into `main/wiki`

### More complex: Retype sources (.md files, configuration) in another repo

I will only describe this one, as it would require more knowledge about github actions and GitHub personal access token (PAT).

Basically, place Retype actual sources (config file and md docs) in a separate repo, and replace the `action-github-pages` step completely with custom code to copy the assembled Retype page (the previous step, `action-build`, exports an environment variable with the path where files are in); and then you'd need to use the GitHub PAT to clone + move files (you can simply erase the `wiki` folder, place over build files, then commit + push back.

This one would be particularly challenging because, as the changes should go across different repos, further authentication needs to take place to send to other repos.

I have tried to be as brief and detailed as possible; if you choose an option and have trouble going thru it, let us know. The "simple", first option may be suggested as a pull request to your repo, the other two will require action from repo owner. (create branch, integrate with other repo)
